airflow/jobs/scheduler_job.py

```diff
@@ -581,6 +593,13 @@ def _process_executor_events(self, session: Session = None) -> int:
                 self.log.info("Setting external_id for %s to %s", ti, info)
                 continue
 
+            # update the last_scheduling_decision for all task in this dag
+            session.query(TI).filter(TI.dag_id == ti.dag_id, TI.state == State.SCHEDULED,).update(
+                {
+                    TI.last_scheduling_decision: timezone.utcnow(),
```

Comment:
```suggestion
            # reset the last_scheduling_decision for all task in this dag
            session.query(TI).filter(TI.dag_id == ti.dag_id, TI.state == State.SCHEDULED).update(
                {
                    TI.last_scheduling_decision: None,
```

Don't we want to set them to null here? How does this interplay with the mini scheduler loop?